src/main/java/org/jboss/threads/EnhancedQueueExecutor.java

```diff
@@ -1795,6 +1801,12 @@ private int tryExecute(final Runnable runnable) {
                     break;
                 }
                 assert tr == AT_NO;
+                // see if we need to spin
+                if (spins > 0) {
+                    JDKSpecific.onSpinWait();
+                    spins--;
+                    continue;
+                }
```

Comment:
Actually, does this interact poorly with growth resistance? I think this effectively invalidates non-zero resistance values.

Reply:
onSpinWait while holding tailLock probably isn't what we want to do in this case. I'm not confident spinning on submit provides any benefit over additional spins before parking.

Reply:
You're right, that (and spinning) is only useful without a lock.  I wonder if it's generally worth unlocking (regardless) before calling `tryAllocateThread`.